### PR TITLE
[ci] Update macOS to 11

### PR DESCRIPTION
**Pull request by strega-nil**
Major new things:
* update to macOS Big Sur (11.2.3)
* switch from VirtualBox to Parallels due to ^ (and also ARM)

Minor new things:
* update from xcode CLT 12 to 12.4

### CODE REVIEW

**Comment by cenit** [comment]
there are not just regressions, but a lot of "CASCADED_DUE_TO_MISSING_DEPENDENCIES".
Please do not overlook them because they were checked previously in CI (vtk, qt, opencv, ...)

**Comment by NancyLi1013** [comment]
`sdformat6 `failed with the following error:
```
CMake Error at /usr/local/lib/cmake/Boost-1.75.0/BoostConfig.cmake:240 (if):
  if given arguments:

    "ALL" "IN_LIST" "Boost_FIND_COMPONENTS"

  Unknown arguments specified
Call Stack (most recent call first):
  /Users/vagrant/Data/work/1/s/scripts/buildsystems/vcpkg.cmake:828 (_find_package)
  /Users/vagrant/Data/downloads/tools/cmake-3.19.2-osx/cmake-3.19.2-macos-universal/CMake.app/Contents/share/cmake-3.19/Modules/FindBoost.cmake:460 (find_package)
  cmake/SearchForStuff.cmake:16 (include)
  CMakeLists.txt:116 (include)
```
`forest `failed with wrong hash due to the repository doesn't exist now. Please see https://github.com/microsoft/vcpkg/pull/16836. 

Billy also set the related triplets as fail in ci.baseline.txt in PR #17331.

File: scripts/azure-pipelines/osx/Setup-VagrantMachines.ps1 [hunk 1]
```diff
@@ -92,33 +82,43 @@ if (-not [String]::IsNullOrEmpty($Date)) {
...
 
+if ([String]::IsNullOrEmpty($MachineId)) {
+    $hostname = hostname -s
+    if ($hostname -match '^VCPKGMM-([0-9]{2})$') {
+        $MachineId = $matches[1]
+    } else {
+        Write-Error "Hostname ($hostname) does not match the expected format (VCPKGMM-NN). Please pass -MachineId in order to give the VM a number."
+    }
+}
+
```
Comment (BillyONeal) [inline scripts/azure-pipelines/osx/Setup-VagrantMachines.ps1:93]: 👍

File: scripts/azure-pipelines/osx/Setup-VagrantMachines.ps1 [hunk 2]
```diff
@@ -92,33 +82,43 @@ if (-not [String]::IsNullOrEmpty($Date)) {
...
 if (Test-Path '~/vagrant/vcpkg-eg-mac') {
-    if ($Force) {
+    Push-Location '~/vagrant/vcpkg-eg-mac'
+    try {
         Write-Host 'Deleting existing directories'
-        Remove-Item -Recurse -Force -Path '~/vagrant/vcpkg-eg-mac' | Out-Null
-    } else {
-        throw '~/vagrant/vcpkg-eg-mac already exists; try re-running with -Force'
+        vagrant destroy -f
+        Remove-Item -Recurse -Force -LiteralPath '~/vagrant/vcpkg-eg-mac' | Out-Null
```
Comment (BillyONeal) [inline scripts/azure-pipelines/osx/Setup-VagrantMachines.ps1:99]: Is there ever any concern about this following symlinks or mount points or something and deleting the underlying base box?

Asking for a friend 🤣

**Review by BillyONeal — APPROVED** [review]

File: scripts/azure-pipelines/osx/Setup-VagrantMachines.ps1 (hunk 2 above)
Comment (strega-nil-ms) [reply scripts/azure-pipelines/osx/Setup-VagrantMachines.ps1:99]: there is not, because it removes `vagrant/vcpkg-eg-mac`, never `vagrant` itself.

**Comment by NancyLi1013** [comment]
`sdformat6 `failed with the same error as before on osx.

`cppgraphqlgen `failed with the following error on osx:
```
FAILED: src/CMakeFiles/graphqlpeg.dir/GraphQLTree.cpp.o 
/Library/Developer/CommandLineTools/usr/bin/c++  -I/Users/vagrant/Data/buildtrees/cppgraphqlgen/src/v3.5.0-ff5eccf561.clean/src/../include -I/Users/vagrant/Data/buildtrees/cppgraphqlgen/src/v3.5.0-ff5eccf561.clean/src/../PEGTL/include -isystem /Users/vagrant/Data/installed/x64-osx/include -fPIC -O3 -DNDEBUG -arch x86_64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk -std=gnu++17 -MD -MT src/CMakeFiles/graphqlpeg.dir/GraphQLTree.cpp.o -MF src/CMakeFiles/graphqlpeg.dir/GraphQLTree.cpp.o.d -o src/CMakeFiles/graphqlpeg.dir/GraphQLTree.cpp.o -c /Users/vagrant/Data/buildtrees/cppgraphqlgen/src/v3.5.0-ff5eccf561.clean/src/GraphQLTree.cpp
clang: error: unable to execute command: Abort trap: 6
clang: error: clang frontend command failed due to signal (use -v to see invocation)
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: x86_64-apple-darwin20.3.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
clang: note: diagnostic msg: PLEASE submit a bug report to http://developer.apple.com/bugreporter/ and include the crash backtrace, preprocessed source, and associated run script.
```
Other failures caused by `gdk-pixbuf` will be fixed in PR #17528.

**Review by NancyLi1013 — APPROVED** [review]

**Pull request merged**